packages/amplify-category-notifications/src/multi-env-manager.ts

```diff
@@ -204,103 +339,97 @@ const pushChanges = async (context: $TSContext, pinpointNotificationsMeta: $TSMe
-    backendConfig[AmplifyCategories.NOTIFICATIONS] = backendConfig[AmplifyCategories.NOTIFICATIONS] || {};
-
-    const resources = Object.keys(backendConfig[AmplifyCategories.NOTIFICATIONS]);
-    for (const resource of resources) {
-      const serviceMeta = backendConfig[AmplifyCategories.NOTIFICATIONS][resource];
-      if (serviceMeta.service === AmplifySupportedService.PINPOINT) {
-        delete backendConfig[AmplifyCategories.NOTIFICATIONS][resource];
-      }
-    }
-
-    if (pinpointMeta) {
-      backendConfig[AmplifyCategories.NOTIFICATIONS][pinpointMeta.serviceName] = {
-        service: pinpointMeta.service,
-        channels: pinpointMeta.channels,
-      };
-    }
+const pushChanges = async (context: $TSContext, pinpointNotificationsMeta: $TSAny):Promise<Array<IChannelAPIResponse|undefined>> => {
+  let pinpointInputParams : $TSAny;
 
-    const jsonString = JSON.stringify(backendConfig, null, 4);
-    fs.writeFileSync(backendConfigFilePath, jsonString, 'utf8');
+  if (context?.exeInfo?.inputParams?.categories?.[AmplifyCategories.NOTIFICATIONS]?.[AmplifySupportedService.PINPOINT]
+  ) {
+    pinpointInputParams = context.exeInfo.inputParams.categories[AmplifyCategories.NOTIFICATIONS][AmplifySupportedService.PINPOINT];
+    context.exeInfo.pinpointInputParams = pinpointInputParams;
   }
-};
 
-const writeAmplifyMeta = (context: $TSContext, categoryMeta: $TSMeta, amplifyMetaFilePath: string) => {
-  if (fs.existsSync(amplifyMetaFilePath)) {
-    const amplifyMeta = context.amplify.readJsonFile(amplifyMetaFilePath);
-    amplifyMeta[AmplifyCategories.NOTIFICATIONS] = categoryMeta;
-    const jsonString = JSON.stringify(amplifyMeta, null, '\t');
-    fs.writeFileSync(amplifyMetaFilePath, jsonString, 'utf8');
+  const pinpointAppStatus = await getPinpointAppStatus(
+    context,
+    context.exeInfo.amplifyMeta,
+    pinpointNotificationsMeta,
+    context.exeInfo.localEnvInfo.envName,
+  );
+  await ensurePinpointApp(context, pinpointNotificationsMeta, pinpointAppStatus, context.exeInfo.localEnvInfo.envName);
+
+  const tasks: Array<$TSAny> = [];
+  const results: Array<IChannelAPIResponse | undefined> = [];
+
+  /**
+   * per current understanding, the following code is only executed when the user is in pull and env add states.
+   * In the Pull/Env add case input params are empty and need to be initialized from the context.exeInfo.backendConfig
+   */
+  if (!pinpointInputParams && context.exeInfo.amplifyMeta[AmplifyCategories.NOTIFICATIONS]) {
+    pinpointInputParams = buildPinpointInputParameters(context);
   }
+  const { channelsToEnable, channelsToDisable } = getEnabledDisabledChannelsFromConfigAndMeta(pinpointInputParams,
+    pinpointNotificationsMeta);
+
+  channelsToEnable.forEach(channel => tasks.push(async () => {
+    try {
+      const result = await notificationManager.enableChannel(context, channel);
+      results.push(result);
+    // eslint-disable-next-line no-empty
+    } catch (err) {}
```

Comment:
Instead of disabling linter can we do something like this
```
try {
  foo();
} catch (err) {
  // explain why it's ok to swallow error
}
```
??

Reply:
I removed the try/catch, I don't think the initEnv should succeed if we fail to enable the channel